solr/core/src/java/org/apache/solr/servlet/PrometheusMetricsServlet.java

```diff
@@ -0,0 +1,139 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.servlet;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.lang.management.MemoryUsage;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import com.sun.management.UnixOperatingSystemMXBean;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.output.CloseShieldOutputStream;
+ /**
+ * FullStory: a simple servlet to produce a few stats formatted for Prometheus.
+ */
+public final class PrometheusMetricsServlet extends BaseSolrServlet {
+   public static void writePromDoc(PrintWriter writer, String name, String purpose, String type) {
+     writer.printf("# HELP %s %s", name, purpose);
+     writer.println();
+     writer.printf("# TYPE %s %s", name, type);
+     writer.println();
+   }
+   @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+     Writer out = null;
+    try {
+      response.setCharacterEncoding("UTF-8");
+      response.setContentType("application/json");
+       // Protect container owned streams from being closed by us, see SOLR-8933
+      out = new OutputStreamWriter(new CloseShieldOutputStream(response.getOutputStream()), StandardCharsets.UTF_8);
+      writeStats(new PrintWriter(out));
+    } finally {
+      IOUtils.closeQuietly(out);
+    }
+  }
+   static void writeStats(PrintWriter pw) {
+    Metrics statz = new Metrics();
+    statz.write(pw);
+    pw.flush();
+  }
+   static class Metrics {
+    Map<String, GarbageCollectorMXBean> gc;
+    MemoryUsage heap;
+    MemoryUsage nonHeap;
+    UnixOperatingSystemMXBean osBean;
+
+    Metrics() {
+      this.gc = new LinkedHashMap<>();
+      for (GarbageCollectorMXBean gcBean : ManagementFactory.getGarbageCollectorMXBeans()) {
+        this.gc.put(gcBean.getName(), gcBean);
+      }
+
+      MemoryMXBean memoryBean = ManagementFactory.getMemoryMXBean();
+      this.heap = memoryBean.getHeapMemoryUsage();
+      this.nonHeap = memoryBean.getNonHeapMemoryUsage();
+      this.osBean = (UnixOperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+    }
+    
+    public void write(PrintWriter writer) {
+      // G1 Young Generation
+      GarbageCollectorMXBean youngGcBean = this.gc.get("G1 Young Generation");
```

Comment:
this seems super sketchy to me..  when I the statz test on my local system I see:

```
  "gc":{
    "PS Scavenge":{
      "collectionCount":0,
      "collectionTime":0},
    "PS MarkSweep":{
      "collectionCount":0,
      "collectionTime":0}},
```

Local dev solr shows:

```
  "gc":{
    "G1 Young Generation":{
      "collectionCount":5,
      "collectionTime":57},
    "G1 Old Generation":{
      "collectionCount":0,
      "collectionTime":0}},
```

https://solr-c01-1.internal.staging.fullstory.com/solr/admin/statz (also prod) shows:

```
    "ParNew":{
      "collectionCount":19658,
      "collectionTime":1715758},
    "ConcurrentMarkSweep":{
      "collectionCount":7,
      "collectionTime":7370}},
```